Fixed reading progress indicator for very short articles

djordjevlais opened the pull request:
ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

- When an entire article fits into the viewport height, we used to show`0%` in the reading progress indicator. Now we check if that's the case, and then show `100%` if it is.

djordjevlais committed:
Fixed reading progress indicator for very short articles

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

- When an entire article fits into the viewport height, we used to show`0%` in the reading progress indicator. Now we check if that's the case, and then show `100%` if it is.

mike182uk commented on apps/admin-x-activitypub/src/components/feed/ArticleModal.tsx:
```diff
@@ -527,17 +533,29 @@ const ArticleModal: React.FC<ArticleModalProps> = ({
             const articleRect = article.getBoundingClientRect();
             const containerRect = container.getBoundingClientRect();
+
+            // Calculate if content is shorter than viewport
```
I don't think you need to comment this, the code is self explanatory 👍

mike182uk commented on apps/admin-x-activitypub/src/components/feed/ArticleModal.tsx:
```diff
@@ -527,17 +533,29 @@ const ArticleModal: React.FC<ArticleModalProps> = ({
+                return;
+            }
+
+            // Existing progress calculation for longer content
```
I don't think you need to comment this, the code is self explanatory 👍

mike182uk commented on apps/admin-x-activitypub/src/components/feed/ArticleModal.tsx:
```diff
@@ -547,9 +565,26 @@ const ArticleModal: React.FC<ArticleModalProps> = ({
-        return () => container?.removeEventListener('scroll', handleScroll);
-    }, []);
+        // Only set up observers and listeners once content is loaded
+        if (!isLoading) {
```
Could maybe return early here instead so we don't have another level of nesting:

```
if (isLoading) {
  return;
}

// ...
```

djordjevlais committed:
Added early return to avoid nesting

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

djordjevlais committed:
Debounced the scroll handler for better performance

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

djordjevlais committed:
Added early return for clearer code

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

djordjevlais committed:
Added a named constant for progress increment

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll

- This way it doesn't seem that number 5 is simply a magic number, it's purpose is more clear and easier to update if needed.

djordjevlais committed:
Bumped the package

ref https://linear.app/ghost/issue/AP-653/scroll-percentage-remains-at-0percent-when-no-content-to-scroll
